var/spack/repos/builtin/packages/intel-mkl/package.py

```diff
@@ -83,6 +83,34 @@ def blas_libs(self):
     def lapack_libs(self):
         return self.blas_libs
 
+    @property
+    def scalapack_libs(self):
+        libnames = [ 'libmkl_scalapack' ]
+        if self.spec.satisfies('^openmpi'):
+            libnames.append('libmkl_blacs_openmpi')
+        elif self.spec.satisfies('^mpich@1'):
+            libnames.append('libmkl_blacs')
+        elif self.spec.satisfies('^mpich@2:'):
+            libnames.append('libmkl_blacs_intelmpi')
```

Comment:
I've rechecked the mkl link line advisor for many different versions of MKL and this appears to be accurate. MKL has used intelmpi for MPICH2 (and 3) and Intel MPI for a long time.
Besides, there's no ```libmkl_blacs_mpich_lp64.so``` in my MKL lib directory.